apps/web/src/app/app/(dashboard)/[workspaceSlug]/settings/_components/plan.tsx

```diff
@@ -0,0 +1,157 @@
+"use client";
+
+import { useTransition } from "react";
+import { useRouter } from "next/navigation";
+import { Check } from "lucide-react";
+import type { z } from "zod";
+
+import type { selectWorkspaceSchema } from "@openstatus/db/src/schema/workspace";
+
+import { Shell } from "@/components/dashboard/shell";
+import { Button } from "@/components/ui/button";
+import { getStripe } from "@/lib/stripe/client";
+import { cn } from "@/lib/utils";
+import { api } from "@/trpc/client";
+
+interface Plan {
+  title: string;
+  description: string;
+  cost: number | string;
+  features: string[];
+  action: {
+    text: string;
+    onClick: () => void;
+  };
+  loading?: boolean;
+}
+
+interface Props extends Plan {
+  className?: string;
+}
+export const SettingsPlan = ({
+  workspaceSlug,
+  workspaceData,
+}: {
+  workspaceSlug: string;
+  workspaceData: z.infer<typeof selectWorkspaceSchema>;
+}) => {
+  const router = useRouter();
+
+  const [isPending, startTransition] = useTransition();
+
+  const getCheckoutSession = () => {
+    startTransition(async () => {
+      const result = await api.stripeRouter.getCheckoutSession.mutate({
+        workspaceId: workspaceSlug,
+      });
+      if (!result) return;
+
+      const stripe = await getStripe();
+      stripe?.redirectToCheckout({ sessionId: result.id });
+    });
+  };
+  const getUserCustomerPortal = async () => {
+    const url = await api.stripeRouter.getUserCustomerPortal.mutate({
+      workspaceId: workspaceSlug,
+    });
+    if (!url) return;
+    router.push(url);
+    return;
+  };
+  getUserCustomerPortal;
+  const plans: Record<"free" | "pro", Plan> = {
+    free: {
+      title: "Free",
+      description: "Get started now and upgrade once reaching the limits.",
+      cost: 0,
+      features: [
+        "5 monitors",
+        "1 status page",
+        "subdomain",
+        "10m, 30m, 1h checks",
+      ],
+      action: {
+        text: workspaceData?.plan === "free" ? "Current plan" : "Downgrade",
+        onClick: async () => {
+          await getUserCustomerPortal();
+        },
+      },
+    },
+    pro: {
+      title: "Pro",
+      description: "Scale and build monitors for all your services.",
+      cost: 29,
+      features: [
+        "20 monitors",
+        "5 status page",
+        "custom domain",
+        "1m, 5m, 10m, 30m, 1h checks",
+        "5 team members",
+      ],
+      loading: isPending,
+      action: {
+        text: workspaceData?.plan === "free" ? "Upgrade" : "Current plan",
+        onClick: async () => {
+          await getCheckoutSession();
+        },
+      },
+    },
+  };
```

Comment:
Fine for now. Once we do a polish, we should extract the data probably inside of the `@openstatus/plans` package. We could overwrite the action inside of the file later on. Something like

```js
import { features } from "@openstatus/plans";
const plans = {
  free: {
    ...features.free,
    action: {
      text: workspaceData?.plan === "free" ? "Upgrade" : "Current plan",
      onClick: async () => (await getCheckoutSession())
    }
  }
}
```